### PR TITLE
Chore/node lts

### LINKED ISSUE #2036: 🚀 Feature: Upgrade node version to 16.13.1 (LTS)
### 🔖 Feature description

Upgrade node version to node `16.13.1` (LTS)

### 🎤 Pitch

The `16.x` release line moved lately (2021-10-26)  into "Active LTS" and will remain so until October 2022.

### ✌️ Solution

Upgrade node version to node `16.13.1` (LTS)

1. make sure all packages pass install, build, test and run
2. do a sanity tests to validate functionality is not broken
3. update push validation github strategy to validate pushes against node '14.X' and '16.X'

### 🔄️ Alternative

not relevant

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/amplication/amplication/blob/master/code_of_conduct.md)

### PR DESCRIPTION
**Pull request by tupe12334**
resolve #2036

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,8 +36,8 @@
     "ts-jest": "27.0.5"
   },
   "engines": {
-    "node": "^14.0.0",
-    "npm": "^7.3.0"
+    "node": "^16.13.0",
+    "npm": "^8.1.0"
```
Comment (mshidlov) [inline package.json:40]: This is not part of this branch

Comment (tupe12334) [reply package.json:40]: why? we updating the engines to node lts

File: .github/workflows/push-validation.yml [hunk 2]
```diff
@@ -46,19 +45,24 @@ jobs:
...
         with:
           path: |
             ~/.npm
             node_modules
             */*/node_modules
-          key: PUSH-${{ runner.os }}-${{matrix.package.package}}-${{ hashFiles(format('{0}/package-lock.json',matrix.package.path)) }}
+          key: PUSH-${{ runner.os }}-${{matrix.package.package}}-${{matrix.node-version}}-${{ hashFiles(format('{0}/package-lock.json',matrix.package.path)) }}
           restore-keys: |
-            PUSH-${{ runner.os }}-${{matrix.package.package}}
+            PUSH-${{ runner.os }}-${{matrix.package.package}}-${{matrix.node-version}}
```
Comment (mshidlov) [inline .github/workflows/push-validation.yml:65]:
```suggestion
            PUSH-${{ runner.os }}-${{matrix.node-version}}-${{matrix.package.package}}-
            PUSH-${{ runner.os }}-${{matrix.node-version}}-
```

File: .github/workflows/setup-script-test.yml [hunk 3]
```diff
@@ -10,13 +10,12 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, windows-latest, ubuntu-latest]
```
Comment (mshidlov) [inline .github/workflows/setup-script-test.yml:12]:
```suggestion
        os: [macos-latest, windows-latest, ubuntu-latest]
        node: [14.x,16.x]
```

File: .github/workflows/setup-script-test.yml [hunk 4]
```diff
@@ -10,13 +10,12 @@ jobs:
...
-        node: [14]
     steps:
       - uses: actions/checkout@v2
       - name: Use Node.js
         uses: actions/setup-node@v2
         with:
-          node-version: 14.x
+          node-version: 16.x
```
Comment (mshidlov) [inline .github/workflows/setup-script-test.yml:18]:
```suggestion
          node-version: ${{ matrix.node}}
```

File: package.json [hunk 5]
```diff
@@ -36,7 +36,7 @@
     "ts-jest": "27.0.5"
   },
   "engines": {
-    "node": "^14.0.0",
+    "node": "^16.13.0",
```
Comment (mshidlov) [inline package.json:39]: We are still supporting node 14 so I don't think this change is needed

File: package.json [hunk 6]
```diff
@@ -36,7 +36,7 @@
     "ts-jest": "27.0.5"
   },
   "engines": {
-    "node": "^14.0.0",
+    "node": "^16.0.0 || ^14.0.0",
```
Comment (mshidlov) [inline package.json:39]:
```suggestion
    "node": "^16.3.1 || ^14.0.0",
```

**Review by mshidlov — APPROVED** [review]

**Pull request merged**